LDAR_Sim/model_code/input_manager.py

```diff
@@ -0,0 +1,216 @@
+        self.method_defaults = {
+            'aircraft': copy.deepcopy(default_aircraft_parameters),
+            'OGI': copy.deepcopy(default_OGI_parameters),
+            'OGI_FU': copy.deepcopy(default_OGI_FU_parameters),
+            'satellite': copy.deepcopy(default_satellite_parameters),
+            'truck': copy.deepcopy(default_truck_parameters),
+            'continuous': copy.deepcopy(default_continuous_parameters),
+        }
+        return
+
+    def read_and_validate_parameters(self, parameter_filenames):
+        """Method to read and validate parameters
+        :param parameter_filenames: a list of paths to parameter files
+        :return returns fully validated parameters dictionary for simulation in LDAR-Sim
+        """
+        raw_parameters = self.read_parameter_files(parameter_filenames)
+        self.parse_parameters(raw_parameters)
+
+        # Coerce all paths to absolute paths prior to release, add extra guards for other parts of the code
+        # that concatenate strings to construct file paths, and expect trailing slashes
+        self.simulation_parameters['wd'] = os.path.abspath(self.simulation_parameters['wd']) + '//'
+        self.simulation_parameters['output_directory'] = \
+            os.path.abspath(self.simulation_parameters['output_directory']) + '//'
+        return(copy.deepcopy(self.simulation_parameters))
+
+    def write_parameters(self, filename):
+        """Method to write simulation parameters to the file system
+        :param filename: filename to write the parameters to
+        """
+        with open(filename, 'w') as f:
+            f.write(yaml.dump(self.simulation_parameters))
+
+    def read_parameter_files(self, parameter_filenames):
+        """Method to read a collection of parameter files and perform any mapping prior to validation.
+        :param parameter_filenames: a list of paths to parameter files
+        :return returns a list of parameter dictionaries
+        """
+        # Read in the parameter files
+        new_parameters_list = []
+        for parameter_filename in parameter_filenames:
+            new_parameters_list.append(self.read_parameter_file(parameter_filename))
+
+        # Perform any mapping, optionally accumulating mined global parameters
+        global_parameters = {}
+        for i in range(len(new_parameters_list)):
+            new_parameters_list[i], mined_global_parameters = self.map_parameters(new_parameters_list[i])
+            global_parameters.update(mined_global_parameters)
+
+        # Append the mined global parameters for installation after all other parameter updates
+        if len(global_parameters) > 0:
+            global_parameters['parameter_level'] = 'global'
+            new_parameters_list.append(global_parameters)
+
+        return(new_parameters_list)
+
+    def read_parameter_file(self, filename):
+        """Method to read a single parameter file from a filename. This method can be extended to address different
```

Comment:
Watch out for line lengths, PEP8 with max-line 100 characters.

Reply:
Thanks. Fixed.